tests/test_msg_nano_signtx.py

```diff
@@ -0,0 +1,310 @@
+        self.assertEqual(nano.encode_balance(4440329590121742105910495447534801366), '03572d26b8163ca8016a76280cb011d6'.decode('hex'))
+        self.assertEqual(nano.encode_balance(340282366920938463463374607431768211455), 'ffffffffffffffffffffffffffffffff'.decode('hex'))
+
+    def test_block_1(self):
+        self.setup_mnemonic_nopin_nopassphrase()
+        res = self.client.nano_sign_tx(
+            'Nano', NANO_ACCOUNT_0_PATH,
+            link_hash='2e3249f1ffc09608d369e01a701bf03bd05509fab262086a59d09994d315e840'.decode('hex'),
+            representative=REP_OFFICIAL_1,
+            balance=9624176000000000000000000000000,
+        )
+        self.assertIsInstance(res, proto.NanoSignedTx)
+        self.assertEqual(res.block_hash, '517565abb71bdccf03754421b1bcaee8327cfce7a571a844ae5392e851531ece'.decode('hex'))
+        self.assertEqual(res.signature, 'e5449bd97ffcd555b435f1289ebf28be4ec4bc58915f7fbdda7283f6727b134b8fb1cdde4eafcb166f2a8a6642000ef3a088661d4100cd819d8bdddb64dd3a00'.decode('hex'))
+
+    def test_block_2(self):
+        self.setup_mnemonic_nopin_nopassphrase()
+        res = self.client.nano_sign_tx(
+            'Nano', NANO_ACCOUNT_0_PATH,
+            parent_link='2e3249f1ffc09608d369e01a701bf03bd05509fab262086a59d09994d315e840'.decode('hex'),
+            parent_representative=REP_OFFICIAL_1,
+            parent_balance=9624176000000000000000000000000,
+            representative=REP_NANODE,
+            balance=9624176000000000000000000000000,
+        )
+        self.assertIsInstance(res, proto.NanoSignedTx)
+        self.assertEqual(res.block_hash, '1a3ec7d5d246aa987d99fde40ff3cadb8833941391611ec9125014d7458ac406'.decode('hex'))
+        self.assertEqual(res.signature, '992619296a0ffe80bfbf4f48739894c8288dde4dbb7f39225a644b4352acdec58e19e0899a18ffc68d89973438d5e6dd6a77e5ae707f03bc0c6544085175e30e'.decode('hex'))
+
+    def test_block_3(self):
+        self.setup_mnemonic_nopin_nopassphrase()
+        res = self.client.nano_sign_tx(
+            'Nano', NANO_ACCOUNT_0_PATH,
+            grandparent_hash='517565abb71bdccf03754421b1bcaee8327cfce7a571a844ae5392e851531ece'.decode('hex'),
+            parent_link='0000000000000000000000000000000000000000000000000000000000000000'.decode('hex'),
+            parent_representative=REP_NANODE,
+            parent_balance=9624176000000000000000000000000,
+            link_hash='4f3d6ce7553bd16d0c03314efeb696dde1b2ae92a28e6346b5ed2cf6a8ff0d8b'.decode('hex'),
+            representative=REP_NANODE,
+            balance=19624176000000000000000000000000,
+        )
+        self.assertIsInstance(res, proto.NanoSignedTx)
+        self.assertEqual(res.block_hash, '2df9eb25f4b7dd2e9174b47c45ab3987db23c38881fd70473ba1c7aff2519d32'.decode('hex'))
+        self.assertEqual(res.signature, 'f0496f49b59a401322334dd7a7676ae4e894efea09be383bf55362e61eef8f526a247a6feefc3bbfa460abfdc6a21b59781fe6c97c4ee91b323868cdf0a4ed0f'.decode('hex'))
+
+    def test_block_4(self):
```

Comment:
Please add:

```
with self.client:
    self.client.set_expected_responses([
    proto.ButtonRequest(...)
    ])
    res = self.client.nano_sign_tx(...
```

as appropriate to the valid ones so we're testing for ButtonRequests too.

Reply:
Resolved with 11659e9